src/Catalog/Dnx/DnxMaker.cs

```diff
@@ -130,12 +130,33 @@ private StorageContent CreateContent(IEnumerable<string> versions)
 
         private async Task DeleteNupkg(Storage storage, string id, string version, CancellationToken cancellationToken)
         {
-            string relativeAddress = string.Format("{1}/{0}.{1}.nupkg", id, version);
+            string relativeAddress = GetRelativeAddressNupkg(id, version);
             Uri nupkgUri = new Uri(storage.BaseAddress, relativeAddress);
             if (storage.Exists(relativeAddress))
             {
                 await storage.Delete(nupkgUri, cancellationToken);
             }
         }
+
+        public static string Normalize(string version)
```

Comment:
Given that we already have some parts of this file that call `NuGetVersion.Parse` directly instead of `NuGetVersion.TryParse`, including a method that calls these functions (https://github.com/NuGet/NuGet.Services.Metadata/blob/ab7eb14a0e07f994614e9dd8f9f3e23cf9a7c44c/src/Catalog/Dnx/DnxMaker.cs#L42), it should be fine to get rid of this method and use `NuGetVersion.Parse` directly.

If you decide to keep this method, then you should add at least one test case that tests invalid versions.